### ISSUE TITLE
Mypy uses wrong flags

### ISSUE THREAD

**Issue by Anton-Constructor** [microsoft/vscode-mypy#145]
Hello,

I've tried to use the built-in mypy version and it doesn't work. It doesn't show any errors - just nothing happens in the file I edit.

I used my own version of mypy for few months, and all works well, but a bit slow - I need to eat for 10+ seconds to se errors in the code:
```js
// I used this option to use my lown mypy
"mypy-type-checker.path": ["mypy"],
```
 
Now decided to try again. And nothing work with the built-in mypy that comes with extension. Looks strange.

Then I tried to dig a bit into the issue and found that used flags don't supported by mypy.

```sh
2023-09-11 13:28:52.142 [info] /Users/anton/.pyenv/versions/3.10.13/envs/core-api/bin/python -m mypy.dmypy --status-file /var/folders/bla-bla-bla.json run -- --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-code --no-pretty --follow-imports=silent --ignore-missing-imports --show-column-numbers --no-pretty --show-error-end /Users/anton/workspace/core-api/app/helper.py
```

Then I decided to run this command manually and have got next error:
```sh
mypy: error: ambiguous option: --show-error-code could match --show-error-codes, --show-error-code-links
```

Changed it to be `--show-error-codes` and then it becomes this:
```sh
dmypy: follow-imports=silent not supported
```

Then removed the `--follow-imports=silent` part. And now seems that I can run the daemon.

To me it seems that you use some old-style flags that aren't supported by the new versions of mypy. I don't know how this extension works for other developers, if it's out of the box has this issue.

May you please fix the issue with flags in the coming version, or somehow show error to the VS Code user about the problem?

**Comment by karthiknadig** [comment]
This is fixed in pre-release.

**Comment by Anton-Constructor** [comment]
> This is fixed in pre-release.

Switched to pre-release. But it doesn't work either. Yes, it has another errors.

```
...
bundled/libs/mypy/fastparse.py", line 531, in from_operator
    op_name = ASTConverter.op_map.get(type(op))
RecursionError: maximum recursion depth exceeded while calling a Python object
```

And 
```
Daemon started
-------------------------------------------------------------------------------
stderr:
/Users/anton/.pyenv/versions/3.10.13/envs/core-api/lib/python3.10/site-packages/sympy/polys/numberfields/resolvent_lookup.py: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.5.0
```

And it ends with these logs lines:
```
2023-09-11 19:49:17.843 [info] mypy
2023-09-11 19:49:17.843 [info] CWD Server: /Users/anton/workspace/core-api
2023-09-11 19:49:17.965 [info] usage: mypy [-h] [-v] [-V] [more options; see below]
            [-m MODULE] [-p PACKAGE] [-c PROGRAM_TEXT] [files ...]
mypy: error: Missing target module, package, files, or command.

2023-09-11 19:49:17.965 [info] 
```

Seems that you at least do not use dot to scan all files for mypy. You need to use it like this `mypy .` (with a final dot to indicate the current folder).

**Comment by q-wertz** [comment]
Had the same issue as described in the top post. Switching to pre-release (`v2023.3.12551007`) helped in my case...

The issued command is
```
2023-09-13 01:15:54.895 [info] /home/user/VCS/project/.venv/bin/python3 -m mypy.dmypy --status-file /tmp/.vscode.dmypy_status/status-f5ea989e-38f0-4b97-a49b-943a4979b0c0.json run -- --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --config-file pyproject.toml --show-error-end /home/user/VCS/project
```

**Comment by Anton-Constructor** [comment]
> This should now be fixed. Setting `"mypy-type-checker.preferDaemon": false` gives you the same experiences as running `mypy` from the command line.

With experimental version of mypy it seems work fine and fast. That is surprising to me (I'm on M2 MacBook Pro). the mypy checker in VS Code always worked super slow, but now fast.

I've fe questions:
- Do you use my pre-installed version of mypy from my env, or is it your own version that ships with the extension?
- Does it use my configuration from `pyproject.toml` automatically?
- Do it see all my installed packages, including type stubs, from my environment? (I used `"mypy-type-checker.path": ["mypy"] setting before only to force use of my local mypy that has access to all my dependencies, and I've installed type stubs to have types from external libraries as well)

**Comment by Anton-Constructor** [comment]
> > Do you use my pre-installed version of mypy from my env, or is it your own version that ships with the extension?
> 
> That depends on the setting `mypy-type-checker.importStrategy`. `useBundled` will use mypy from the extension, `fromEnvironment` will take from the environment.
> 
> > Does it use my configuration from pyproject.toml automatically?
> 
> This runs `mypy` or `dmypy` as it would from command line. So it should pick up.
> 
> > Do it see all my installed packages, including type stubs, from my environment? (I used `"mypy-type-checker.path": ["mypy"] setting before only to force use of my local mypy that has access to all my dependencies, and I've installed type stubs to have types from external libraries as well)
> 
> If you select the interpreter from python extension where all the type stubs are installed then it should be able to use it.

Thank you for your replies.

I've set these options in my `settings.json` file:
```js
    // "mypy-type-checker.path": ["mypy"],
    "mypy-type-checker.importStrategy": "fromEnvironment",
    "mypy-type-checker.args": ["--config-file", "pyproject.toml"],
    "mypy-type-checker.preferDaemon": false,
```

Now my mypy checks work super fast. If previously I were waiting for 20 seconds to show the errors, now it runs instantly. Really huge progress, my congratulations! I also asked couple of other developers in my team to switch to the pre-release version and they also told that now I works really fast.

May I ask you about the expected final release of this feature? That we can switch to the stable release of the mypy extension, as we don't want to keep using the pre-release unstable version in our day-to-day work. Once we will install the final version I will try to reply to you here if all works fast the same way as on the pre-release version. Will be waiting from you a message about readiness for the stable version.

**Comment by karthiknadig** [comment]
This will be available in stable highly likely end of next week. Pending any major issues reported on this feature.
